rs/rosetta-api/tests/system_tests/test_cases/search_transactions.rs

```diff
@@ -182,10 +206,214 @@ fn test_search_transactions_by_index() {
+                                )
+                            }));
+                            // Make sure every fetched transaction is unique
+                            assert_eq!(
+                                result.len(),
+                                result.into_iter().collect::<HashSet<_>>().len()
+                            );
+                        }
+
+                        search_transactions_request.account_identifier = Some(
+                            icp_ledger::AccountIdentifier::from(Account {
+                                owner: ic_base_types::PrincipalId::new_anonymous().into(),
+                                subaccount: Some([9; 32]),
+                            })
+                            .into(),
+                        );
+                        let result = traverse_all_transactions(
+                            &env.rosetta_client,
+                            search_transactions_request.clone(),
+                        )
+                        .await;
+                        // If the account does not exist the service should return an empty list
+                        assert_eq!(result.len(), 0);
+                    }
+                });
+                Ok(())
+            },
+        )
+        .unwrap();
+}
+
+#[test]
+fn test_search_transactions_by_operation_type() {
+    let mut runner = TestRunner::new(TestRunnerConfig {
+        max_shrink_iters: 0,
+        cases: *NUM_TEST_CASES,
+        ..Default::default()
+    });
+
+    runner
+        .run(
+            &(valid_transactions_strategy(
+                (*MINTING_IDENTITY).clone(),
+                DEFAULT_TRANSFER_FEE,
+                *MAX_NUM_GENERATED_BLOCKS,
+                SystemTime::now(),
+            )
+            .no_shrink()),
+            |args_with_caller| {
+                let rt = Runtime::new().unwrap();
+                rt.block_on(async {
+                    let env = RosettaTestingEnvironment::builder()
+                        .with_transfer_args_for_block_generating(args_with_caller.clone())
+                        .with_minting_account(MINTING_IDENTITY.sender().unwrap().into())
+                        .build()
+                        .await;
+
+                    if !args_with_caller.is_empty() {
+                        let operations = vec![
+                            "TRANSFER".to_string(),
+                            "MINT".to_string(),
+                            "BURN".to_string(),
+                            "APPROVE".to_string(),
+                        ];
+
+                        let mut search_transactions_request =
+                            SearchTransactionsRequest::builder(env.network_identifier.clone())
+                                .build();
+
+                        for operation in operations.into_iter() {
+                            // We make sure that the service returns the correct number of transactions for each account
```

Comment:
```suggestion
                            // We make sure that the service returns the correct number of transactions for each operation type
```
Also a similar comment to the one for the test above - IIUC, we don't actually check the number of transactions returned for each operation type here.

Reply:
I'll add a check per operation type that counts the ops types in the blocks of the ledger